lib/spack/spack/build_systems/cmake.py

```diff
@@ -154,6 +154,11 @@ def _std_args(pkg):
             define('CMAKE_BUILD_TYPE', build_type),
         ]
 
+        # Spack does not need to consider multilib environments due to each
+        # package getting its own prefix, so avoid usage of CMake's
+        # GNUInstallDirs module detecting multilib platform directories.
+        args.append(define('CMAKE_INSTALL_LIBDIR', 'lib'))
+
```

Comment:
#17126 seems to report a `vtk` issue. Are we sure we want to enforce this on all `CMake` generated packages rather than just on `vtk`?

Reply:
@alalazo I don't know how many other packages would be affected by this, but at least ParaView is hitting the same issue. This would fix it for ParaView and VTK.

Reply:
I personally prefer the more general solution as long as this option is valid for all CMake packages.

Reply:
Any project using `GNUInstallDirs` on a Red Hat-derived or Debian-derived distro is prone to picking `lib64` or `lib/x86_64-linux-gnu` as the install lib directory. This is for multilib and multiarch features in the distros (respectively). Since Spack isn't ready to handle this in dependency paths (`pkg-config`, `LD_LIBRARY_PATH`, etc.), just avoiding the auto-detection is best.

It's fine since Spack puts everything into unique prefixes, so multi{lib,arch} isn't a problem worth considering here.

Reply:
@alalazo Is this ok? Can this be resolved and move forward?

Reply:
@alalazo ping

Reply:
@EthanS94 Apologies, I missed this thread in the stream of notifications. I understand the arguments that @mathstuf brings. From my point of view as long as these questions can be answered with a "yes" I have no objection:

1. Is this worth impacting all `CMakePackage` packages i.e. are we confident this won't bring side effects for other builds and will be useful to more than a handful of packages?
2. If we do this change in the base class is there a reason to impose `LIBDIR` but not [other variables](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) that are documented in `GNUInstallDirs`?

My concern is that a change in `vtk` will impact `vtk` users, a change in `CMakePackage` will pretty much impact everybody.

Reply:
I would check another package that is using the module and see if it ends up with `lib64` on a Red Hat x86_64 host. If so, this patch is better. If not, VTK might be doing something off, but I can't think of what it would be; we use the module pretty much as-is.

For 2, the other variables aren't so distro-dependent. But setting them probably wouldn't be too bad in case some distros do have other defaults that may interfere.

Reply:
@scottwittenburg Is there an easy way to trigger an entire rebuild of the `e4s` pipeline if I push this branch upstream? Having `e4s` build correctly would make me more confident that the change won't affect negatively other software.

Reply:
@alalazo Do you mean forcing each and every packages to be rebuilt from source, regardless of whether this changes its full hash or not?  If so, then running a pipeline pointing at an empty mirror would it.  One easy way to do that would be to create a branch off master in gitlab.spack.io/spack/e4s, where you just change the mirror path, and then you run pipelines against that branch, setting `SPACK_REPO` and `SPACK_REF` environment variables to point to this fork/branch of spack.

I can set it up and link back here if it sounds like a pain.